autogluon/utils/tabular/ml/learner/default_learner.py

```diff
@@ -105,7 +117,120 @@ def _fit(self, X: DataFrame, X_val: DataFrame = None, scheduler_options=None, hy
         time_end = time.time()
         self.time_fit_training = time_end - time_preprocessing_end
         self.time_fit_total = time_end - time_preprocessing_start
-        logger.log(20, f'AutoGluon training complete, total runtime = {round(self.time_fit_total, 2)}s ...')
+        if X_test_unlabeled is not None:
```

Comment:
This should be moved out of _fit. This code can't reasonably be at the end of _fit, but should be an entirely new call to learner. (post-fit call). No need to fit a new feature generator, can simply pass the test data as unlabelled data into the original _fit call and have the feature generator fit with it (identical to the existing functionality of passing in tuning_data without label)

X, X_val, y, y_val can be loaded from cached files similarly to how it is done in trainer.

self.trainer can be re-used, no need to create a new trainer.

Reply:
Actually, are you sure you need X, X_val, y, y_val at all? Does this really need to live in learner? Why not do this in trainer? The X/X_val won't actually be aligned properly at present since X_val is generally None and the split occurs by random seed in trainer. Altering X will alter the rows used as random seed.